### ISSUE TITLE
IGRAC-WHOS connection

### ISSUE THREAD

**Issue by EliVolsch** [kartoza/IGRAC-GGIS#662]
Priority: Medium

Hi Irwan,
This ticket is about the connection between GGMN and the WHOS service. Can you please check the questions sent by Enrico Boldrini via email (I already forwarded it to you)? Thanks!
Cheers,
Claudia

**Comment by clau1313** [comment]
@danangmassandy I just forwarded the questions via email, thank you

**Comment by meomancer** [comment]
1. only OGC SOS version 1.0.0 is supported at the moment (no 2.0.0), is it correct?
2. It seems the following requests have issues, could you check?
https://ggis.un-igrac.org/istsos?service=SOS&request=GetObservation&version=1.0.0&offering=measurement&procedure=bc.mon.59381&observedProperty=urn:ogc:def:parameter:x-igrac:1.0:Level%20Measurement:Water%20level%20elevation%20a.m.s.l.&responseFormat=text/xml;subtype=%22om/1.0.0%22&procedureDescriptionFormat=http://www.opengis.net/sensorML/1.0.1&api-key=724df1163f0a99f81af89f3b446d76281d07d3ac
https://ggis.un-igrac.org/istsos?service=SOS&version=1.0.0&request=getFeatureOfInterest&FeatureOfInterestId=
 urn:ogc:def:feature:x-istsos:1.0:Water well:SM-0246&api-key=724df1163f0a99f81af89f3b446d76281d07d3ac
https://ggis.un-igrac.org/istsos?service=SOS&version=1.0.0&request=getFeatureOfInterest&FeatureOfInterestId=urn:ogc:def:feature:x-istsos:1.0:Water
 well:001N005W21K001S&api-key=724df1163f0a99f81af89f3b446d76281d07d3ac 
https://ggis.un-igrac.org/istsos?service=SOS&version=1.0.0&request=getFeatureOfInterest&FeatureOfInterestId=urn:ogc:def:feature:x-istsos:1.0:Water
 well:ns.mon.832002&api-key=724df1163f0a99f81af89f3b446d76281d07d3ac 

3. is it possible to download a temporal subset? It should be possible, please confirm the following request or provide a correct one:
https://ggis.un-igrac.org/istsos?service=SOS&request=GetObservation&version=1.0.0&offering=measurement&procedure=bc.mon.59381&observedProperty=urn:ogc:def:parameter:x-igrac:1.0:Level%20Measurement:Water%20level%20elevation%20a.m.s.l.&responseFormat=text/xml;subtype=%22om/1.0.0%22&api-key=724df1163f0a99f81af89f3b446d76281d07d3ac&temporalFilter=phenomenonTime,2016-01-01T00:00:00Z/2016-02-01T23:59:59Z
4. Please confirm the following procedure we will use to harvest the available metadata for all the time series:
·                Get Capabilities to obtain the list of feature of interests. E.g. urn:ogc:def:feature:x-istsos:1.0:Water well:bc.mon.59381
·                Get feature of interest: https://ggis.un-igrac.org/istsos?service=SOS&version=1.0.0&request=getFeatureOfInterest&FeatureOfInterestId=urn:ogc:def:feature:x-istsos:1.0:Water well:bc.mon.59381&api-key=724df1163f0a99f81af89f3b446d76281d07d3ac 
In the response there are: 
- temporal extent 
- Location 
- procedure identifier: bc.mon.59381 useful to retrieve the procedure. 
- Get procedure: https://ggis.un-igrac.org/istsos?service=SOS&version=1.0.0&request=DescribeSensor&procedure=bc.mon.59381&outputFormat=text/xml;subtype=%22sensorML/1.0.1%22&api-key=724df1163f0a99f81af89f3b446d76281d07d3ac 

In the response there are: 
- Temporal extent 
- Location 
- Country 
- License 
- System type (e.g. insitu-fixed-point) 
- Sensor type (e.g. measurement) 
- Observed property name 
- Observed property identifier 
- Observed property units 

Are there any other important metadata we are missing? In case please advice how they can be retrieved.

**Comment by danangmassandy** [comment]
@meomancer I have submitted PR to address the issues in the istsos API.

>  It seems the following requests have issues, could you check?

Fix exception when fields are empty.

> is it possible to download a temporal subset? It should be possible, please confirm the following request or provide a correct one:

The temporalFilter is used in version 2.0.0. Below is the correct URL for temporal subset using version 1.0.0:
https://ggis.un-igrac.org/istsos?service=SOS&request=GetObservation&version=1.0.0&offering=measurement&procedure=bc.mon.59381&observedProperty=urn:ogc:def:parameter:x-igrac:1.0:Level%20Measurement:Water%20level%20elevation%20a.m.s.l.&responseFormat=text/xml;subtype=%22om/1.0.0%22&api-key=724df1163f0a99f81af89f3b446d76281d07d3ac&eventtime=2016-01-01T00:00:00Z/2016-02-01T23:59:59Z

> Please check the capabilities response:
I have tried the capabilities API in my local and it took around 4minutes. The response size is 109 MB, so it potentially becomes request timed out when accessing the API without cache. In the PR, I called the generation of get capabilities cache in the uploader.
